lib/mongo_mapper/plugins/keys.rb

```diff
@@ -110,7 +110,7 @@ def to_mongo(instance)
         end
 
         def from_mongo(value)
-          value && (value.instance_of?(self) ? value : load(value))
```

Comment:
Given the signature of `load`, it requires an attributes hash. But `value` can be an instance of an EmbeddedDocument class.